packages/cli/src/util/output/create-output.ts

```diff
@@ -68,112 +77,99 @@ function _createOutput({ debug: debugEnabled = false }: OutputOptions = {}) {
+    this.print(`${chalk.cyan('> Ready!')} ${str}\n`);
+  };
 
-  function success(str: string) {
-    print(`${chalk.cyan('> Success!')} ${str}\n`);
-  }
+  success = (str: string) => {
+    this.print(`${chalk.cyan('> Success!')} ${str}\n`);
+  };
 
-  function debug(str: string) {
-    if (debugEnabled) {
-      log(
+  debug = (str: string) => {
+    if (this.debugEnabled) {
+      this.log(
         `${chalk.bold('[debug]')} ${chalk.gray(
           `[${new Date().toISOString()}]`
         )} ${str}`
       );
     }
-  }
+  };
 
-  function setSpinner(message: string, delay: number = 300): void {
-    spinnerMessage = message;
-    if (debugEnabled) {
-      debug(`Spinner invoked (${message}) with a ${delay}ms delay`);
+  spinner = (message: string, delay: number = 300): void => {
+    this.spinnerMessage = message;
+    if (this.debugEnabled) {
+      this.debug(`Spinner invoked (${message}) with a ${delay}ms delay`);
       return;
     }
-    if (spinner) {
-      spinner.text = message;
+    if (this._spinner) {
+      this._spinner.text = message;
     } else {
-      spinner = wait(message, delay);
+      this._spinner = wait(message, delay);
     }
-  }
+  };
 
-  function stopSpinner() {
-    if (debugEnabled && spinnerMessage) {
-      const msg = `Spinner stopped (${spinnerMessage})`;
-      spinnerMessage = '';
-      debug(msg);
+  stopSpinner = () => {
+    if (this.debugEnabled && this.spinnerMessage) {
+      const msg = `Spinner stopped (${this.spinnerMessage})`;
+      this.spinnerMessage = '';
+      this.debug(msg);
     }
-    if (spinner) {
-      spinner();
-      spinner = null;
-      spinnerMessage = '';
+    if (this._spinner) {
+      this._spinner();
+      this._spinner = null;
+      this.spinnerMessage = '';
     }
-  }
+  };
 
-  async function time<T>(
+  time = async <T>(
     label: string | ((r?: T) => string),
     fn: Promise<T> | (() => Promise<T>)
-  ) {
+  ) => {
     const promise = typeof fn === 'function' ? fn() : fn;
 
-    if (debugEnabled) {
+    if (this.debugEnabled) {
       const startLabel = typeof label === 'function' ? label() : label;
-      debug(startLabel);
+      this.debug(startLabel);
       const start = Date.now();
       const r = await promise;
       const endLabel = typeof label === 'function' ? label(r) : label;
       const duration = Date.now() - start;
       const durationPretty =
         duration < 1000 ? `${duration}ms` : `${(duration / 1000).toFixed(2)}s`;
-      debug(`${endLabel} ${chalk.gray(`[${durationPretty}]`)}`);
+      this.debug(`${endLabel} ${chalk.gray(`[${durationPretty}]`)}`);
       return r;
     }
 
     return promise;
-  }
-
-  return {
-    isDebugEnabled,
-    print,
-    log,
-    warn,
-    error,
-    prettyError,
-    ready,
-    success,
-    debug,
-    dim,
-    time,
-    note,
-    spinner: setSpinner,
-    stopSpinner,
   };
 }
+
+export default function createOutput(opts?: OutputOptions) {
+  return new Output(opts);
```

Comment:
I noticed this is no longer a singleton.

What happens when `createOutput()` is called more than once?

Reply:
The codebase has been cleaned up at this point to not create many output instances, just a single one gets created in the entrypoint file. The singleton requirement was more of a temporary band-aid while cleaning that part up. I think it's ok to remove that requirement at this point.